go/vt/vttablet/tabletmanager/tm_init.go

```diff
@@ -831,6 +837,47 @@ func (tm *TabletManager) handleRestore(ctx context.Context, config *tabletenv.Ta
 	return false, nil
 }
 
+// waitForDBAGrants waits for DBA user to have the required privileges to function properly.
+func (tm *TabletManager) waitForDBAGrants(config *tabletenv.TabletConfig, waitTime time.Duration) (err error) {
+	// We should close the _waitForGrantsComplete channel in the end to signify that the wait for dba grants has completed.
+	defer func() {
+		if err == nil {
+			close(tm._waitForGrantsComplete)
+		}
+	}()
+	// We don't wait for grants if the tablet is externally managed. Permissions
+	// are then the responsibility of the DBA.
+	if config == nil || config.DB.HasGlobalSettings() || waitTime == 0 {
+		return nil
+	}
+	timer := time.NewTimer(waitTime)
+	ctx, cancel := context.WithTimeout(context.Background(), waitTime)
+	defer cancel()
+	for {
+		conn, connErr := dbconnpool.NewDBConnection(ctx, config.DB.DbaConnector())
```

Comment:
We should close the conn in a defer when err is nil.

Reply:
Thats a good catch! I totally forgot about closing the connection! I can't put it in a defer, because we are creating these connections in a loop, but I have added closing the connection right after running the fetch query.